Client/mods/deathmatch/logic/CClientObjectManager.cpp

```diff
@@ -142,23 +92,18 @@ CClientObject* CClientObjectManager::Get(ElementID ID)
 
 bool CClientObjectManager::IsValidModel(unsigned long ulObjectModel)
 {
-    bool bIsValid = false;
-
-    if (ulObjectModel < 20001)
+    if (ulObjectModel < 20000)
     {
-        unsigned int uiChunk = ulObjectModel / (sizeof(unsigned int) * 8);
-        unsigned int shift = ulObjectModel - (uiChunk * sizeof(unsigned int) * 8);
-        unsigned int bit = 1 << shift;
-        bIsValid = !!(g_iValidObjectModels[uiChunk] & bit);
-
-        // Return whether we can grab the model information or not
-        if (bIsValid && !g_pGame->GetModelInfo(ulObjectModel))
+        CModelInfo* pModelInfo = g_pGame->GetModelInfo(ulObjectModel);
+        if (pModelInfo && pModelInfo->GetInterface())
         {
-            bIsValid = false;
+            eModelInfoType eType = pModelInfo->GetModelType();
+            if (eType == eModelInfoType::CLUMP || eType == eModelInfoType::ATOMIC || eType == eModelInfoType::WEAPON)
+                return true;
         }
     }
 
-    return bIsValid;
+    return false;
 }
```

Comment:
Since this entire function has essentially been rewritten, please can you make use of early returns?

(E.g. `if ((x > 2000) || (x < 0)) { return false; }`  etc ... except formatted correctly)